Fix CI build by using pre-built rootfs image

fangn2 opened the pull request:
*Issue #, if available:*
#418 
The main branch is failing due to hardcoded rootfs name as the name changed with upstream firecracker moving from bionic to jammy.  Also devtool used to build rootfs is unstable. 
```
cp: cannot stat 'build/firecracker/build/rootfs/bionic.rootfs.ext4': No such file or directory
```

*Description of changes:*
1. Not specify the ubuntu version name but using regex to match it. The approach is safe since we will only have one rootfs built at a time.
2. Use prebuilt rootfs and ssh key from S3 instead of building from devtool on fly which seems pretty unstable.
3.  Update readme of snapshotting example to reflect current status. 
4.  Minor update for the snapshotting example code to clean up generated files. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
Signed-off-by: Tony Fang <nhfang@amazon.com>

swagatbora90 commented on Makefile:
```diff
@@ -110,7 +110,7 @@ $(FC_TEST_DATA_PATH)/root-drive-ssh-key $(FC_TEST_DATA_PATH)/root-drive-with-ssh
 	$(MAKE) $(FIRECRACKER_DIR)
 	$(FIRECRACKER_DIR)/tools/devtool build_rootfs -m $(FC_TEST_DATA_PATH)/mnt
-	cp $(FIRECRACKER_DIR)/build/rootfs/bionic.rootfs.ext4 $(FC_TEST_DATA_PATH)/root-drive-with-ssh.img
+	cp $(FIRECRACKER_DIR)/build/rootfs/*.rootfs.ext4 $(FC_TEST_DATA_PATH)/root-drive-with-ssh.img
```
This looks good for now. We can also make a change to `/tools/devtool` to build the rootfs drive with a custom name provided as an argument.

fangn2 replied:
The tricky part is the devtool is owned by [firecracker](https://github.com/firecracker-microvm/firecracker/blob/main/tools/devtool#L1160). We need to request changes upstream to do that.

fangn2 replied:
Anyway,  decided to bypass the devtool which seems unstable, and uses pre-built rootfs instead following suggestion from #418.

swagatbora90 reviewed (approved):
LGTM. The buildkite go-sdk and go-sdk-amd CI steps  have failed. PTAL before merging.

swagatbora90 commented on examples/cmd/snapshotting/Makefile:
```diff
@@ -50,28 +50,27 @@ bin/host-local: bin
-	cp temp/build/rootfs/ssh/id_rsa root-drive-ssh-key
-	rm -rf temp
-
+	curl -L -o root-drive-with-ssh.img https://s3.amazonaws.com/spec.ccfc.min/ci-artifacts/disks/${ARCH}/ubuntu-18.04.ext4
```
`devtool` is using ubuntu 22.04, correct? We should keep the same upgraded version here imo.

fangn2 replied:
Current HEAD of firecracker uses 22.04. Older versions of firecracker use 18.04.
Current issue is firecracker didn't publish `ubuntu-18.04.ext4` in the same S3 bucket. 
I can follow up with them and address this in a follow-up PR?

fangn2 commented on .buildkite/pipeline.yml:
```diff
@@ -61,11 +61,9 @@ steps:
-      # Download Firecracker and its jailer.
-      - make deps
-      # Build a rootfs with SSH enabled.
-      - sudo -E FC_TEST_DATA_PATH=${FC_TEST_DATA_PATH} make ${FC_TEST_DATA_PATH}/root-drive-ssh-key
```
With pre-built rootfs, we no longer need to build them with root, I added them as part of deps build.

fangn2 commented on Makefile:
```diff
@@ -132,12 +125,14 @@ $(FC_TEST_BIN_PATH)/tc-redirect-tap:
 	$(call install_go,github.com/kunalkushwaha/ltag,v0.2.3)
 
-$(FIRECRACKER_DIR):
-	- git clone https://github.com/firecracker-microvm/firecracker.git $(FIRECRACKER_DIR)
```
Move this block down to put related code together for better readability.

fangn2 commented:
Updated few other parts(please check Description of changes for details) to pass the build.  
Cleaned up the examples code base a little bit. 

CI seems a little bit unstable when it ran step `test against Firecracker above` which uses firecracker built from HEAD of main, but I think we can monitor from there. 

@swagatbora90  @austinvazquez  @ginglis13  PTAL

swagatbora90 commented on Makefile:
```diff
@@ -42,17 +42,13 @@ $(FC_TEST_DATA_PATH)/ltag \
-endif
+$(FC_TEST_BIN_PATH)/tc-redirect-tap \
+$(FC_TEST_DATA_PATH)/root-drive-with-ssh.img \
+$(FC_TEST_DATA_PATH)/root-drive-ssh-key
```
nit: alphabetize the order of test object. the new targets moved after L41?

fangn2 replied:
Updated.

swagatbora90 commented on examples/cmd/snapshotting/README.md:
````diff
@@ -62,20 +62,5 @@ Pressing enter resumes execution of the program.
-```
-sudo rm -rf root-drive-with-ssh.img root-drive-ssh-key
-sudo make image
+make clean
````
what was changed here that we no longer need sudo?

fangn2 replied:
Devtool used to build rootfs uses docker --privileged, then generated rootfs file are under root, needs sudo to copy to test location. 
The new approach directly download from S3 no longer needs that.

swagatbora90 reviewed (approved):
Minor nits, otherwise looks good.

We are using `hello-rootfs.ext4` for non-ssh testing, which I believe is also uploaded by firecracker team. Lets confirm what OS version this is based of and whether FC team can do something similar for ssh version, i.e. with OS agnostic naming.
